project/SparkBuild.scala

```diff
@@ -1001,9 +1004,9 @@ object TestSettings {
     javaOptions += "-Xmx3g",
     // Exclude tags defined in a system property
     testOptions in Test += Tests.Argument(TestFrameworks.ScalaTest,
-      sys.props.get("test.exclude.tags").map { tags =>
-        tags.split(",").flatMap { tag => Seq("-l", tag) }.toSeq
-      }.getOrElse(Nil): _*),
+      sys.props.get("test.exclude.tags").map(tag => tag.split(",").toSeq)
+        .getOrElse(defaultExcludedTagsForScalaTest).filter(!_.trim.isEmpty)
```

Comment:
This looks not robust enough.
If a developer set an irrelevant tag (`ExtendedSQLTest` or `ExtendedHiveTest`), `ChromeUITest` suddenly will cause a failure in the irrelevant `Core` module.

Reply:
It's what I intended that if developers set `test.excluded.tags` explicitly, all the tags which are intended to be excluded should be added explicitly.
But as you concerned, this change seems to break the build compatibility.
I'll submit another idea.